src/layers/microscopy-viewer-layer/microscopy-viewer-layer-base.js

```diff
@@ -40,51 +38,42 @@ const defaultProps = {
 
 export class MicroscopyViewerLayerBase extends BaseTileLayer {
   constructor(props) {
-    const { sliderValues, colorValues, channelsOn } = props;
-    const orderedChannelNames = Object.keys(sliderValues);
-    const { orderedSliderValues, orderedColorValues } = setOrderedValues(
-      orderedChannelNames,
-      colorValues,
-      sliderValues,
-      channelsOn
+    const { sliderValues, colorValues, channelsOn, loader } = props;
+
+    const lengths = [sliderValues.length, colorValues.length];
+    if (lengths.every(l => l !== lengths[0])) {
+      throw Error('Inconsistent number of slider values and colors provided');
+    }
+
+    const colors = colorValues.map((color, i) =>
+      channelsOn[i]
```

Comment:
Maybe rename to `channelIsOn`? Or keep, the name, but change it to a set?

Reply:
Agreed. Generally I think we should remove `channelsOn`/`channelIsOn` all together. I wasn't sure if that was within the scope of this PR. Right now when we turn a channel "off" we still fetch all the tiles related to that channel but make the color `[0,0,0]` and slider `[max_slider_val, max_slider_val]`. We should only fetch tiles for the number of channels expected to be returned from `getTile`. The UI component should be responsible for:

1.) Managing which channels are returned from `loader.getTile`.
2.) the colors associated with these channels.
3.) the slider values associated with these channels.